### PR TITLE
Add teacher mode

### LINKED ISSUE #299: teacher mode
Implement teacher mode:
- statistics are neither recorded nor shown
- all lessons and steps are unlocked
- "to current step" button is unavailable

                

When the user returns back to learner mode, he/she continues from the last step he visited before turning on teacher mode.

### PR DESCRIPTION
**Pull request by winter-yuki**
Closes #299

### CODE REVIEW

**Comment by winter-yuki** [comment]
I'm going to save the current lesson of the user's session and restore it after disabling the teacher mode

**Comment by winter-yuki** [comment]
My emulator simultaneously writes messages that "System UI is not responding" and suggest closing the app. Is it an emulator issue or of the app, @zuevval ?

**Comment by zuevval** [comment]
> My emulator simultaneously writes messages that "System UI is not responding" and suggest closing the app. Is it an emulator issue or of the app, @zuevval ?

@winter-yuki Are you still facing this problem? I've compiled this version of the app and launched it on a physical device + three different emulators, and everything works fine.
Moreover, I've run the instrumental tests (`./gradlew connectedCheck`) and they've passed, so it seems strange that CI is failing.

**Comment by winter-yuki** [comment]
> > My emulator simultaneously writes messages that "System UI is not responding" and suggest closing the app. Is it an emulator issue or of the app, @zuevval ?
> 
> @winter-yuki Are you still facing this problem? I've compiled this version of the app and launched it on a physical device + three different emulators, and everything works fine.
> Moreover, I've run the instrumental tests (`./gradlew connectedCheck`) and they've passed, so it seems strange that CI is failing.

It can be my  IOS emulator issue, so thank you for the check.

File: app/src/main/java/com/github/braillesystems/learnbraille/ui/screens/settings/SettingsFragment.kt [hunk 1]
```diff
@@ -1,11 +1,102 @@
...
+        preference.onPreferenceChangeListener =
+            Preference.OnPreferenceChangeListener { _, newValue ->
+                require(preference is SwitchPreferenceCompat)
+                require(newValue is Boolean)
+
+                val saveCurrStepPreference =
+                    "save_curr_step_teacher_mode_${preferenceRepository.currentUserId}"
+                if (newValue) {
+                    askCode {
+                        val code = 8436
```
Comment (zuevval) [inline app/src/main/java/com/github/braillesystems/learnbraille/ui/screens/settings/SettingsFragment.kt:43]: Would you mind replacing it with a more memorable `0001`?

**Comment by zuevval** [comment]
> It can be my IOS emulator issue, so thank you for the check.

Note: I've just built the app on macOS and launched it on  a virtual Pixel 3, it works

<details>
  <summary>screenshot</summary>

Screenshot 2021-08-24 at 16 20 28

</details>

File: app/src/main/java/com/github/braillesystems/learnbraille/ui/screens/settings/SettingsFragment.kt (hunk 1 above)
Comment (winter-yuki) [reply app/src/main/java/com/github/braillesystems/learnbraille/ui/screens/settings/SettingsFragment.kt:43]: It should not be so memorable for students, I guess :)

Comment (zuevval) [reply app/src/main/java/com/github/braillesystems/learnbraille/ui/screens/settings/SettingsFragment.kt:43]: As you wish, let it be `8436`

**Comment by winter-yuki** [comment]
Screenshot 2021-08-24 at 23 50 35

It sometimes appears after some use. But there are other stranges too: scrolling does not work properly, it rolls back. So it's hard to me to enable the new preference in settings.

**Comment by winter-yuki** [comment]
> It sometimes appears after some use. But there are other stranges too: scrolling does not work properly, it rolls back. So it's hard to me to enable the new preference in settings.

After pressing "Wait", app works as usual.

**Comment by zuevval** [comment]
@winter-yuki I've noticed, too, that scrolling does not work as I expect on my MBP. Unfortunately did not find any advice on the internet (only [a question on a forum](https://askandroidquestions.com/2021/03/25/android-emulator-two-finger-scroll-laggy-on-macbook/))

**Comment by zuevval** [comment]
In the last commit, I've slightly modified the descriptions of the new settings item and changed the dialog buttons' colors (to match the rest of the app).
screenshot

**Pull request merged**